pkg/provisioner/crdprovisioner.go

```diff
@@ -517,6 +517,7 @@ func (c *CrdProvisioner) PublishVolume(
 			return nil, err
 		}
 		// if Replica AzVolumeAttachment CRI does not exist for the volume-node pair for failover, create a new Primary
+		// aliceyu??? why replica? it could be primary does not exist
```

Comment:
`CrdProvisioner.PublishVolume` is called to for the primary disk attachment. If an `AzVolumeAttachment` already exists, we assume that it is replica attachment and just need to promote it.

Reply:
ah thanks! I forgot to delete my own notes!